src/tox/config/source/discover.py

```diff
@@ -58,9 +69,10 @@ def _create_default_source(root_dir: Path | None) -> Source:
     if root_dir is None:  # if set use that
         empty = Path.cwd()
         for base in chain([empty], empty.parents):
-            if (base / "pyproject.toml").exists():
-                empty = base
-                break
+            for file_path in SOURCE_TYPES:
+                if (base / file_path.FILENAME).exists():
```

Comment:
This change is unrelated 🤔 what's this about?

Reply:
I think that was another bug I found while testing because it only checked for presence of `pyproject.toml` file, when in fact it should look for all the 3 supported filenames, in particular order from SOURCE_TYPES.

Basically before this it was looking on parents only for pyproject.toml.

Reply:
Do we have a test to validate this then? The one you added I think only deals with the folder parts of it.

Reply:
For some reason I do not see any codecov job reporting coverage. That would make it much easier to decide.